bigtable-dataflow-import/src/main/java/com/google/cloud/bigtable/dataflowimport/HBaseResultToMutationFn.java

```diff
@@ -48,25 +81,78 @@ static void setLogger(Logger log) {
   @Override
   public void processElement(ProcessContext context) throws Exception {
     KV<ImmutableBytesWritable, Result> kv = context.element();
-    Cell[] cells = checkEmptyRow(kv);
-    if (cells.length == 0) {
+    List<Cell> cells = checkEmptyRow(kv);
+    if (cells.isEmpty()) {
       return;
     }
-    // TODO: Handle DeleteMarkers properly.
-    Put put = new Put(kv.getKey().get());
+    Put put = tryProcessRowWithNoDeleteMarkers(kv.getKey().get(), cells);
```

Comment:
I think using the optimistic try makes it easy to tell that the no-marker path has the least overhead possible.